model/proto/api_v2.proto

```diff
@@ -64,6 +64,28 @@ service CollectorService {
     }
 }
 
+message GetTraceRequest {
+  bytes trace_id = 1 [
+    (gogoproto.nullable) = false,
+    (gogoproto.customtype) = "github.com/jaegertracing/jaeger/model.TraceID",
+    (gogoproto.customname) = "TraceID"
+  ];
+}
+
+message GetTraceResponseStream {
+  repeated jaeger.api_v2.Span spans = 1 [
+    (gogoproto.nullable) = false
+  ];
+}
+
+service QueryService {
+    rpc GetTrace(GetTraceRequest) returns (stream GetTraceResponseStream) {
+        option (google.api.http) = {
+            post: "/api/v2/traces/{trace_id}"
```

Comment:
This doesn't seem right to me. The `trace_id` should either be part of the body like so - 

```
service QueryService {
    rpc GetTrace(GetTraceRequest) returns (stream GetTraceResponseStream) {
        option (google.api.http) = {
            post: "/api/v2/traces"
            body: "*"
        };
    }
}
```

...or, we could make it a string parameter.

```
message GetTraceRequest {
  string trace_id = 1 [
    (gogoproto.nullable) = false,
    (gogoproto.customtype) = "github.com/jaegertracing/jaeger/model.TraceID",
    (gogoproto.customname) = "TraceID"
  ];
}
```

Reply:
if it is `bytes` as in the main API, then
  * how would swagger file describe the parameter?
  * how would it look on the wire?

I don't have a strong opinion about post vs. get here. Get seems more REST-ful.

Reply:
Changing it to bytes and generating swagger documentation gives - 
```
.
.
    "/api/v2/traces": {
      "post": {
        "operationId": "GetTrace",
        "responses": {
          "200": {
            "description": "(streaming responses)",
            "schema": {
              "$ref": "#/definitions/api_v2GetTraceResponseStream"
            }
          }
        },
        "parameters": [
          {
            "name": "body",
            "in": "body",
            "required": true,
            "schema": {
              "$ref": "#/definitions/api_v2GetTraceRequest"
            }
          }
        ],
        "tags": [
          "QueryService"
        ]
      }
    }
  },
.
.
  "definitions": {
    "api_v2GetTraceRequest": {
      "type": "object",
      "properties": {
        "trace_id": {
          "type": "string",
          "format": "byte"
        }
      }
    },
.
.
```

Reply:
https://swagger.io/docs/specification/data-models/data-types/#string

`"type": "string","format": "byte"` would mean base64 encoded traceID on the wire

Reply:
I think it’s the same for ids in the returned span JSON, so sounds reasonable?

Reply:
+1

> I don't have a strong opinion about post vs. get here. Get seems more REST-ful.

You're right, I should change this to `get`.